### DISCUSSION TITLE
Genotype calling for an autotetraploid species without reference genome

### DISCUSSION THREAD

**Discussion by jorisbertrand** [lvclark/polyRAD#19] [Q&A]
Dear Lindsay, 
first of all we would like to thank you for the promising initiative that PolyRAD offers for population geneticists wanting to deal with natural populations of non-model polyploid species.
We would like to use PolyRAD to call genotypes from an autotetrapoloid species (106 individuals genotyped through a nGBS protocol from 9 localities). 
SNPs were called with STACKS (v.2.41) in de novo mode (as we do not have any reference genome available) for this species. As far as we know, the populations display a significant level of genetic structure and moderate to low degrees of genetic diversity.
PolyRAD seems very complete and what we want to do is rather quite simple: input data from STACKS, infer the most likely genotypes and output the results to feed other population genomics softwares.
The appropriate plan seems to follow what you showed in the video tutorial: i) quality check and filtering, ii) use the IteratePopStruct() function to infer genotypes and iii) export the results as a .vcf (or a structure file).
If we are correct, we may want to have a quick feedback on the values we obtained at different steps but before that, we would like to know whether or not it is possible to output RADdata objects (that have been called with the readStacks() function) as a vcf. We did some tests and got the following error message:
Error in RADdata2VCF(mydataPopStruct, "test.vcf") : 
  Complete haplotype information not provided; unable to determine SNP positions.  Use refgenome argument in VCF2RADdata.
We also tried to Export_Structure(). It works but we are not sure to understand the values of the genotypes as displayed in the file. 
All the best,

**Comment by lvclark** [comment]
Hi Joris,

I'm glad that you're finding polyRAD helpful!  VCFs have columns for chromosome and position, so if that information is not available, polyRAD will not export to VCF.  As I recall with the Stacks import option, the full tag sequences are not imported, just variable nucleotides, so that's something to keep in mind as well when interpreting allele names.

The Structure file should have four rows for each individual, and one column for each locus.  Each allele is assigned a number, and the most probable genotype is shown.  So if you have 1,1,2,3, that means two copies of one allele, one copy of a different allele, and then one copy of a third allele.

Best,

Lindsay

**Reply by jorisbertrand** [reply]
Dear Lindsay, 
thank you so much for your answer.
OK so now we have a .str file that we would like to use as input for downstream analysis (e.g. genind object). The think is that neither the adegenet::import2genind nor adegenet::read.structure() are able to import this STRUCTURE file because individual genotypes are encoded on more than two consecutive rows. We also tried to convert the file with PGDSpider (http://www.cmpg.unibe.ch/software/PGDSpider/) but got similar issue.
Later on, we noticed that you authored a couple of years ago, a package called 'polysat' that include a read.Structure(), a write.Genodive() or a gendata.to.genind() that may do exactly what we want. Unfortunately, the read.Structure() function does not work with our file. R runs forever (about two days) and nothing happens.
So I am sorry again but I have a very naive question of bad-coder evolutionary ecologists. Do you see any alternative to make our output of PolyRAD readable by classical population genetic softwares?
If you want to have a look, I attach the corresponding .str file (I added a .txt extension otherwise the file could not be uploaded to Github).
All the best,
Joris
[test_bis.str.txt](https://github.com/lvclark/polyRAD/files/6141505/test_bis.str.txt)

**Reply by lvclark** [reply]
Dear Joris,

I don't recommend polysat for performing the conversion because (1) it was designed for a much smaller number of markers than is typically found in a GBS dataset, hence why it was running slowly, and (2) it assumes allele copy number ambiguity, and would throw away polyRAD's allele copy number estimates.

Not every classical population genetic software is going to allow polyploid genotypes, so you'll have to be specific about which software you want to use.   Is your end goal here to perform analysis in adegenet?  I just looked at the documentation to remind myself of the format of the `genind` class, and I think it would be pretty quick for me to make an export function from polyRAD.  (In fact, I'm sure the data structures of polyRAD were inspired by adegenet, consciously or unconsciously.)

Best,

Lindsay

**Reply by jorisbertrand** [reply]
Hi Lindsay,
Yes! adegenet would definitely be an option to perform analyses such as PCA or DAPC. Also being able to import the data in Genodive could be great as this software is able to perform polyploid dosage corrections. I guess including such features in PolyRAD could allow to make it more broadly used in our field.
Just let me know if you need some more information about our data.
Thank you for your precious help,
Cheers
Joris

**Reply by lvclark** [reply]
FYI, polyRAD does PCA, and the results are available in the `$PCA` slot.  One could also use `GetWeightedMeanGenotypes` to generate a matrix that could be passed to the `adegenet::find.clusters` and `adegenet::dapc` functions, bypassing the need to import as `genind`.

**Reply by lvclark** [reply]
And if importing from polyRAD, the procedure for correction of allele dosages in polyploids in GenoDive would be unneccessary, since polyRAD has already estimated allele dosages using a similar procedure.  Supposedly GenoDive can import from Structure format, although I'm not sure if it would require one to add a column indicating population assignments.  If it doesn't work, let me know.  A GenoDive export function would be more involved for me to create than a genind export function would be, and I'm working on a grant this week so I don't want to make more work for myself than is necessary.

**Reply by lvclark** [reply]
Dear Joris,

I just added a `Export_adegenet_genind` function to the version of polyRAD on GitHub.  If you get a chance to test it out this week, let me know how it goes!  I need to update polyRAD on CRAN very soon anyway, so if there are any tweaks needed for the new function I might as well include them.

Best,

Lindsay

**Reply by jorisbertrand** [reply]
Thanks a lot! we are gonna try. Also we managed to open the structure file in GenoDive. By default, it was automatically trying to open it as a 'Migrate IAM' file, so we add to import it as a text file, change the format to 'Structure' and then use the menu Data > Transformation to set up the ploidy level. Keep in touch!
